lib/options.nix

```diff
@@ -333,5 +333,109 @@ rec {
+                submodule {
+                  freeformType = attrsOf anything;
+                  options = {
+                    # Spec loading:
+                    enabled = mkStrLuaFnOr types.bool (lazyLoadDefaults.enabledInSpec or null) ''
+                      When false, or if the function returns false, then ${originalName} will not be included in the spec.
+
+                      Equivalence: lz.n => enabled; lazy.nvim => enabled
+                    '';
+
+                    priority = mkNullable types.number (lazyLoadDefaults.priority or null) ''
+                      Only useful for start plugins (not lazy-loaded) to force loading certain plugins first.
+
+                      Equivalence: lz.n => priority; lazy.nvim => priority
+                    '';
+
+                    # Spec setup
+                    # Actions
+                    beforeAll = mkLuaFn (lazyLoadDefaults.beforeAll or null) ''
+                      Always executed before any plugins are loaded.
+
+                      Equivalence: lz.n => beforeAll; lazy.nvim => init
+                    '';
+
+                    before = mkLuaFn (lazyLoadDefaults.before or null) ''
+                      Executed before ${originalName} is loaded.
+
+                      Equivalence: lz.n => before; lazy.nvim => None
+                    '';
+
+                    after = mkLuaFn (lazyLoadDefaults.after or null) ''
+                      Executed after ${originalName} is loaded.
+
+                      Equivalence: lz.n => after; lazy.nvim => config
+                    '';
+
+                    # Triggers
+                    event = mkNullable triggerType (lazyLoadDefaults.event or null) ''
+                      Lazy-load on event. Events can be specified as `BufEnter` or with a pattern like `BufEnter *.lua`
+
+                      Equivalence: lz.n => event; lazy.nvim => event
+                    '';
+
+                    cmd = mkNullable triggerType (lazyLoadDefaults.cmd or null) ''
+                      Lazy-load on command.
+
+                      Equivalence: lz.n => cmd; lazy.nvim => cmd
+                    '';
+
+                    ft = mkNullable triggerType (lazyLoadDefaults.ft or null) ''
+                      Lazy-load on filetype.
+
+                      Equivalence: lz.n => ft; lazy.nvim => ft
+                    '';
+
+                    keys = mkListOf (types.attrsOf types.anything) (lazyLoadDefaults.keys or null) ''
+                      Lazy-load on key mapping.
+
+                      Equivalence: lz.n => keys; lazy.nvim => keys
+                    '';
+
+                    colorscheme = mkNullable triggerType (lazyLoadDefaults.colorscheme or null) ''
+                      Lazy-load on colorscheme.
+
+                      Equivalence: lz.n => colorscheme; lazy.nvim => None
+                    '';
+                  };
+                };
+            };
+          };
+        };
+      default = lazyLoadDefaults;
```

Comment:
This feels wrong. If we have defaults they should probably be set on a per-option basis, either when declaring the option or using `lib.mkOptionDefault`.

Generally, the top-level default should be `{}` or `[]`